pkg/ssh/ssh.go

```diff
@@ -97,6 +101,12 @@ func NewClient(cfg *Config, logger log.Logger, km *KeyManager) *Client {
 func (s *Client) starting(ctx context.Context) error {
 	level.Info(s.logger).Log("msg", "starting ssh client")
 
+	if !s.cfg.SkipSSHValidation {
+		if err := validateSSHVersion(ctx, s.SSHCmd); err != nil {
+			return fmt.Errorf("failed to validate ssh version: %w", err)
```

Comment:
I interpret this message as "the validation check failed to complete", which is true sometimes. I think we should distinguish between the two cases (failed to complete validation; warn and continue && validation completed and failed; error)

Reply:
Sounds reasonable. I refactored the code a bit to follow this, warn if unable to properly fetch the version and error if the version is deemed invalid.